plugins/inputs/prometheus/prometheus.go

```diff
@@ -46,6 +50,37 @@ type Prometheus struct {
 	kubernetesPods map[string]URLAndAddress
 	cancel         context.CancelFunc
 	wg             sync.WaitGroup
+
+	// List of consul services to scrape
+	consulServices map[string]URLAndAddress
+}
+
+type ConsulConfig struct {
+	// Address of the Consul agent. The address must contain a hostname or an IP address
+	// and optionally a port (format: "host:port").
+	Enabled       bool              `toml:"enabled"`
+	Agent         string            `toml:"agent,omitempty"`
+	QueryInterval internal.Duration `toml:"query_interval"`
+	Queries       []ConsulQuery     `toml:"query"`
+}
+
+// One Consul service discovery query
+type ConsulQuery struct {
+	// A name of the searched services (not ID)
+	ServiceName string `toml:"name"`
+
+	// A tag of the searched services
+	ServiceTag string `toml:"tag"`
+
+	// A template URL of the Prometheus gathering interface. The hostname part
+	// of the URL will be replaced by discovered address and port.
+	ServiceURL string `toml:"url"`
+
+	// Extra tags to add to metrics found in Consul
+	ServiceExtraTags map[string]string `toml:"tags"`
+
+	serviceURLTemplate       *template.Template
+	serviceExtraTagsTemplate map[string]*template.Template
 }
```

Comment:
It might make sense to move these declarations into `consul.go` as they are mostly used there.